src/algebra/pointwise.lean

```diff
@@ -593,6 +598,7 @@ instance set_semiring.add_comm_monoid : add_comm_monoid (set_semiring α) :=
   add_zero := union_empty,
   add_comm := union_comm, }
 
+@[to_additive set_semiring.non_unital_non_assoc_add_semiring]
```

Comment:
I'm very confused about what this does - surely this puts a second addition and multiplication on `set_semiring`?

Reply:
Eric, you are probably right that this is not what I had in mind and indeed it broke other things.

I will ask in Zulip, but what I want to achieve is that the operation used on the sets is addition instead of multiplication.  My use-case is subsets of the naturals, where addition is union and multiplication of sets is the set of all possible *sums* of natural numbers, rather than their products.

Reply:
That's `set_semiring (multiplicative ℕ)`

Reply:
Ah, thank you!